src/components/Dashboard/index.js

```diff
@@ -73,283 +127,488 @@ const Dashboard = () => {
     [dispatch]
   );
 
-  const onSubmit = async ({
-    name,
-    handle,
-    country,
-    org_handle,
-    org_name,
-    org_website,
-    org_country,
-  }) => {
-    setError("");
-    await setUpInitialData({
-      orgData: showOrgForm,
-      name,
-      handle,
-      country,
-      org_handle,
-      org_name,
-      org_website,
-      org_country,
-    })(firebase, firestore, dispatch);
+  const onSubmit = async () => {
+    validateHandle().then(async (validateHandle) => {
+      if (showOrgForm) {
+        validateOrgHandle().then(async (validateOrgHandle) => {
+          validateCountry();
+          validateOrgCountry();
+          validateName();
+          validateOrgName();
+          validateOrgWebsite();
```

Comment:
Be creative here. It's fine to have this way but it would be better if this can be moved into a separate function.

Reply:
I will move them to a separate function and commit. Thank you for mentioning that.